common/lib/xmodule/xmodule/templates/html/sgp.html

```diff
@@ -0,0 +1,52 @@
+<h4>{{display_name}}</h4>
+<hr>
+{{instructions|safe}}
+
+{% if score.score %}
+<h3>Your grade: <strong>{{score.grade_pct}}</strong></h3>
+{% else %}
+{% if grades_available %}
+<p><em>Not yet scored</em></p>
+{% else %}
+<p><em>View in LMS to see score</em></p>
+{% endif %}
+{% endif %}
+{% if is_staff and grades_available %}
+<div class="row">
+    <div class="col">
+        <hr>
+        <p>Please upload a CSV with the grades. Thanks. Have a nice day.</p>
+        <form action="{{import_url}}" method="POST" enctype="multipart/form-data">
+            <input type="hidden" name="csrfmiddlewaretoken" value="{{csrf_token}}">
+            <p>
+                <a href="{{export_url}}" class="btn btn-brand">download CSV</a>
+                <input id="{{id}}-file" type="file" name="csv">
+                <a href="#" id="{{id}}-import" class="submit btn btn-brand">import CSV</a>
+                <h2 id="{{id}}-status" class="text-info" style="display:none"></h2>
+            </p>
+        </form>
+    </div>
+</div>
+<script type="text/javascript">
```

Comment:
I know there's a way for us to put our js through webpack, and we're obligated to via some oep or another.

I'll look into this more; can you bug me if I don't follow up quickly enough with more actionable feedback on this point?

Reply:
According to a quick discussion with Cale I'm way off base on expecting webpack to be an easy "oh that's already supported" situation. I'm marking that portion as resolved for now; I may look into it a bit more but I definitely know enough to say webpack shouldn't block us here as of right now.

Only possible question is could we pass the JS as a string to e.g. the `add_javascript` function in e.g. https://edx.readthedocs.io/projects/xblock-tutorial/en/latest/concepts/fragments.html#fragments-and-views ? It wouldn't put stuff through webpack, but it'd still be a bit better than a straight-up script tag.

Reply:
Yes, the JS can be in its own file. It'd require some refactoring, obviously. https://github.com/edx/xblock-lti-consumer/blob/master/lti_consumer/lti_consumer.py#L738

I'm sure you have a better sense of how to make this JS less gnarly. We should talk about it.